src/components/SideBar.jsx

```diff
@@ -14,58 +14,91 @@ import {
+  const [showAll, setShowAll] = useState(true);
+  const itemsArray = Object.keys(items).map((item) => items[item]);
+
   const handleOnClick = (event) => {
-    console.log(event);
+    let output = itemsArray.filter(
+      (item) => item.vault_collection_name == event.target.outerText
+    );
+    setFilterItems(output);
+    setShowAll(false);
+    setCollectionName(event.target.outerText);
   };
 
   return (
-    <Box sx={{ display: "flex" }}>
-      <CssBaseline />
-      <AppBar
-        position="fixed"
-        sx={{ zIndex: (theme) => theme.zIndex.drawer + 1 }}
-      >
-        <NavBar />
-      </AppBar>
-      <Drawer
-        variant="permanent"
-        sx={{
-          width: drawerWidth,
-          flexShrink: 0,
-          [`& .MuiDrawer-paper`]: {
+    <>
+      <Box sx={{ display: "flex" }}>
+        <CssBaseline />
+        <AppBar
+          position="fixed"
+          sx={{ zIndex: (theme) => theme.zIndex.drawer + 1 }}
+        >
+          <NavBar />
+        </AppBar>
+        <Drawer
+          variant="permanent"
+          sx={{
             width: drawerWidth,
-            boxSizing: "border-box",
-          },
-        }}
-      >
+            flexShrink: 0,
+            [`& .MuiDrawer-paper`]: {
+              width: drawerWidth,
+              boxSizing: "border-box",
+            },
+          }}
+        >
+          <Toolbar />
+          <Box sx={{ overflow: "auto", paddingLeft: 2 }}>
+            <List>
+              <Typography sx={{ fontSize: "h5.fontSize" }}>
+                Collections
+              </Typography>
+              <div>
+                {collections.map((collections) => {
+                  return (
+                    <ListItem key={collections.name}>
+                      <ListItemButton onClick={(event) => handleOnClick(event)}>
+                        <ListItemText primary={collections.name} />
+                      </ListItemButton>
+                    </ListItem>
+                  );
+                })}
+              </div>
+            </List>
+            <Divider />
+          </Box>
+        </Drawer>
         <Toolbar />
-        <Box sx={{ overflow: "auto", paddingLeft: 2 }}>
-          <List>
-            <Typography sx={{ fontSize: "h5.fontSize" }}>
-              Collections
-            </Typography>
-            <div>
-              {collections.map((collections) => {
-                return (
-                  <ListItem key={collections.name}>
-                    <ListItemButton onClick={handleOnClick}>
-                      <ListItemText primary={collections.name} />
-                    </ListItemButton>
-                  </ListItem>
-                );
-              })}
-            </div>
-          </List>
-          <Divider />
-        </Box>
-      </Drawer>
-      <Toolbar />
-    </Box>
+      </Box>
+      <Box sx={{ paddingLeft: 35 }}>
+        <>
+          {showAll ? (
+            <VaultItemTiles items={items} setItems={setItems} />
```

Comment:
I'm seeing some inconsistency in the data structures you're passing to `VaultItemTiles`.

`items` is the original Object from `DashboardPage`, and you're passing it in here. But on line 95 you're passing an Array instead.

I think it would make most sense to always pass it an Array, and change `VaultItemTiles` to expect an Array.

On this line you could just do

```js
<VaultItemTiles items={Object.values(items)} setItems={setItems} />
```

instead of having a separate itemsArray state variable